raiden/tests/integration/transfer/test_mediatedtransfer_events.py

```diff
@@ -40,22 +41,43 @@ def test_mediated_transfer_events(raiden_network, token_addresses, network_wait)
         to_identifier='latest',
     )
     initiator_events = [blocknumber_event[1] for blocknumber_event in initiator_blockevents]
-    assert must_contain_entry(initiator_events, SendRevealSecret, {})
-    assert must_contain_entry(initiator_events, EventUnlockSuccess, {})
+    assert wait_until(
+        lambda: must_contain_entry(initiator_events, SendRevealSecret, {}),
```

Comment:
**none** of the calls to `wait_until` on this PR are correct, they are effectively only adding a sleep and don't change the result of the `assert` expression.

All occurrences are using closures over a list which is *never* reevaluated (e.g. the `initiator_events` above), therefor the truthiness or falseness of the call to `must_contain_entry` will never change.

There is really no need to use higher-order functions for this, just write a new function in the `raiden.waiting` module if you need something.

Reply:
Hmm, you're right.. weirdly, the tests passed after adding it, so I think it's just a matter of timing, as I originally thought. [tonight's build](https://travis-ci.org/raiden-network/raiden/jobs/398024107#L1733) failed because of this, I'll fix it to re-evaluate the lists in the loop, it should address this comment and properly fix the test.

Reply:
Done in #1688